### PR TITLE
perldelta for ae78ae4a0f2, smart match removal

### PR DESCRIPTION
**Pull request by tonycoz**
Largely copied from the deprecation notice in perl5380delta

### CODE REVIEW

**Review by jkeenan — COMMENTED** [review]
Should we provide some/any guidance to users as to how to replace switch and smartmatch in their code?

**Comment by khwilliamson** [comment]
I'm thinking the same thing.  Damian has something  on cpan that he says does this.  We could just mention that such modules exist instead of blessing a particular implementation

**Comment by jkeenan** [comment]
> I'm thinking the same thing. Damian has something on cpan that he says does this. We could just mention that such modules exist instead of blessing a particular implementation

I guess that would be https://metacpan.org/pod/Switch::Back.  Can readers suggest others?

**Comment by rwp0** [comment]
> > I'm thinking the same thing. Damian has something on cpan that he says does this. We could just mention that such modules exist instead of blessing a particular implementation
> 
> I guess that would be https://metacpan.org/pod/Switch::Back. Can readers suggest others?

I also saw this recent module by @Leont claiming to be a right implementation:

> [Syntax::Infix::Smartmatch](https://metacpan.org/pod/Syntax::Infix::Smartmatch) - Smartmatch done right

**Comment by Grinnz** [comment]
I would add: https://metacpan.org/pod/Syntax::Operator::Matches, and https://metacpan.org/pod/Syntax::Keyword::Match which may eventually become a core feature.

File: pod/perldelta.pod [hunk 1]
```diff
@@ -69,6 +69,31 @@ They were deprecated in Perl v5.38.0 and scheduled for removal in Perl
 
 These features have now been entirely removed.
 
+If you code currently uses C<given>/C<when> or smart match this can be
+replaced with an C<if>/C<else> chain, or there is a variety of
```
Comment (khwilliamson) [inline pod/perldelta.pod]: I think `is` is technically right here, but `are a variety` sounds better to me

Comment (tonycoz) [reply pod/perldelta.pod]: I originally used "are", but I've dodged the problem now.

**Review by jkeenan — APPROVED** [review]

**Comment by Leont** [comment]
> I also saw this recent module by @Leont claiming to be a right implementation:

That is a simpler smartmatch than the one in core, and doesn't provide given/when. It's really only a partial solution.

> I guess that would be https://metacpan.org/pod/Switch::Back. Can readers suggest others?

I think that is a useful idea, but I really hate the implementation. I was planning to write a backwards compatible module that isn't based on trying to parse Perl with regexes but I haven't even gotten around to starting it.

**Comment by tonycoz** [comment]
5.41.3 isn't out just yet, so you can still update perldelta for that release.

Even post release updating `perl5413delta.pod` would help producing the 5.42.0 perldelta.

**Pull request merged**